### PR TITLE
PGD Random Restart Changes

### LINKED ISSUE #714: Change Implementation of Random Restarts in PGD
In it's current implementation PGD with N random restarts runs a batch N times though the PGD attack and picks the batch with the worst performance with the accuracy being computed averaged on a batch basis.

However, this can give overly optimistic scores of model performance. As a simple example, consider two random restarts and a batch size of 10. The first restart the first 5 examples are classified correctly. The second restart the last 5 are classified correctly. In both cases the accuracy is 50% averaged across the batch. However, on a per sample basis (ie the worst score achieved for a given example) the score is 0%.

As an improvement we can change how the accuracy is computed so that is conducted over each sample.

### PR DESCRIPTION
**Pull request by GiulioZizzo**
# Description

We are changing how random restarts are implemented for PGD to select the worst case per example to return, rather then the worst case averaged across the whole datasets.

Fixes #714

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- Accuracy of random restarts is now more effective when testing against various locally trained classifiers under tensorflow and sklearn.

**Test Configuration**:
- OS: RHEL 
- Python version: 3.6 and 3.7
- ART version or commit number: 20ec7f5a526379457ef561ef5a2d2d511cadd1d2
- TensorFlow / Keras / PyTorch / MXNet version: tensorflow==2.3.1, Keras==2.4.3, torch==1.6.0+cu92, mxnet==1.6.0

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by beat-buesser** [comment]
Closing because replaced by #765
